src/safeposix/syscalls/net_calls.rs

```diff
@@ -36,29 +41,45 @@ impl Cage {
         return sockfd;
     }
 
+    //Find an available file descriptor index in the File Descriptor Table
+    //If available, insert the socket fd into the File Descriptor Table at the 
+    //available index
     fn _socket_inserter(&self, sockfd: FileDescriptor) -> i32 {
         let (fd, guardopt) = self.get_next_fd(None);
+        //Indicate an error in the fd and propogate the error forward 
         if fd < 0 {
             return fd;
         }
         let fdoption = &mut *guardopt.unwrap();
-        let _insertval = fdoption.insert(sockfd);
+        //Insert the sockfd as the FileDescriptor inside fdoption
+        let _insertval = fdoption.insert(sockfd); 
         return fd;
     }
 
+    //An implicit bind refers to the automatic binding of a socket to an address 
+    //and port by the system, without an explicit call to the bind() function by the
+    //programmer. This typically happens when the socket is used for client-side 
+    //operations, such as when it initiates a connection to a server.
     fn _implicit_bind(&self, sockhandle: &mut SocketHandle, domain: i32) -> i32 {
         if sockhandle.localaddr.is_none() {
+            //Assign a new local address to the socket handle
             let localaddr = match Self::assign_new_addr(
                 sockhandle,
                 domain,
+                //The SO_RESUEPORT bit within the protocol encodes rebind ability
+                //The rebind ability of a socket refers to whether a socket can be 
+                //re-bound to an address and port that it was previously bound to, 
+                //especially after it has been closed or if the binding has been reset. 
```

Comment:
Okay, why do we set this?  

It's slightly odd that SO_REUSEPORT is used to leftshift a bit.  Why is this done?

Reply:
I found a source that explains the importance of  the SO_RESUSEPORT socket option.

https://lwn.net/Articles/542629/

I was wondering if some post like this is valid to cite since I'm not exactly sure the credentials of the site but it seems legit

Reply:
Sure.  You should cite whatever you looked at.